feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/repository/AccountRepositoryImpl.kt

```diff
@@ -173,9 +173,10 @@ class AccountRepositoryImpl(
     override suspend fun deleteAccount(metaId: Long) = withContext(Dispatchers.Default) {
         val metaAccountType = accountDataSource.getMetaAccountTypeOrThrow(metaId)
 
-        accountDataSource.deleteMetaAccount(metaId)
+        val allAffectedMetaIds = accountDataSource.deleteMetaAccount(metaId)
 
-        metaAccountChangesEventBus.notify(Event.AccountRemoved(metaId, metaAccountType), source = null)
+        val deleteEvents = allAffectedMetaIds.map { Event.AccountRemoved(it, metaAccountType) }
+        metaAccountChangesEventBus.notify(Event.BatchUpdate(deleteEvents), source = null)
```

Comment:
There is an extension `combineBusEvents to convert `List< Event>` to a batch. It also takes into account that there is no need to wrap into a batch if there is just a single event